src/WindowsTracing/GraphicsEtwProvider.cpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "GraphicsEtwProvider.h"
+
+// clang-format off
+#include <guiddef.h>
+#include <stdint.h>
+// clang-format on
+
+#include <PresentData/ETW/Microsoft_Windows_D3D9.h>
+#include <PresentData/ETW/Microsoft_Windows_DXGI.h>
+#include <PresentData/ETW/Microsoft_Windows_Dwm_Core.h>
+#include <PresentData/ETW/Microsoft_Windows_DxgKrnl.h>
+#include <PresentData/ETW/Microsoft_Windows_EventMetadata.h>
+#include <PresentData/ETW/Microsoft_Windows_Win32k.h>
+#include <PresentData/ETW/NT_Process.h>
+
+// clang-format off
+#include <d3d9.h>
+#include <dxgi.h>
+// clang-format on
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/Profiling.h"
+
+namespace orbit_windows_tracing {
+
+using namespace std::placeholders;
+
+GraphicsEtwProvider::GraphicsEtwProvider(uint32_t pid, krabs::user_trace* trace,
+                                         TracerListener* listener)
+    : target_pid_(pid), trace_(trace), listener_(listener) {
+  EnableProvider("Dxgi", Microsoft_Windows_DXGI::GUID,
+                 std::bind(&GraphicsEtwProvider::OnDXGIEvent, this, _1, _2));
+  EnableProvider("D3d9", Microsoft_Windows_D3D9::GUID,
+                 std::bind(&GraphicsEtwProvider::OnD3d9Event, this, _1, _2));
+  EnableProvider("DwmCore", Microsoft_Windows_Dwm_Core::GUID,
+                 std::bind(&GraphicsEtwProvider::OnDwmCoreEvent, this, _1, _2));
+  EnableProvider("DwmCoreWin7", Microsoft_Windows_Dwm_Core::GUID,
+                 std::bind(&GraphicsEtwProvider::OnDwmCoreWin7Event, this, _1, _2));
+  EnableProvider("DxgKrnl", Microsoft_Windows_DxgKrnl::GUID,
+                 std::bind(&GraphicsEtwProvider::OnDxgKrnlEvent, this, _1, _2));
+  EnableProvider("DxgKrnlWin7Pres", Microsoft_Windows_DxgKrnl::Win7::PRESENTHISTORY_GUID,
+                 std::bind(&GraphicsEtwProvider::OnDxgKrnlWin7PresEvent, this, _1, _2));
+  EnableProvider("NtProcess", NT_Process::GUID,
+                 std::bind(&GraphicsEtwProvider::OnNtProcessEvent, this, _1, _2));
+  EnableProvider("WindowsEventMetadata", Microsoft_Windows_EventMetadata::GUID,
+                 std::bind(&GraphicsEtwProvider::OnWindowsEventMetadata, this, _1, _2));
+  EnableProvider("Win32K", Microsoft_Windows_Win32k::GUID,
+                 std::bind(&GraphicsEtwProvider::OnWin32KEvent, this, _1, _2));
```

Comment:
[TotW#108](https://abseil.io/tips/108)
Please, use `absl::bind_front` instead of `std::bind`

Reply:
done, thanks